src/lua_console.h

```diff
@@ -0,0 +1,44 @@
+#ifndef LUA_CONSOLE_H
+#define LUA_CONSOLE_H
+
+#include "output.h"
+#include "input.h"
+
+#include <map>
+#include <string>
+#include <vector>
+#include <utility>
+
+struct WINDOW;
```

Comment:
Don't do this. Include "cursesdef.h" instead. With ncurses, WINDOW is not just a struct. In my versions it's actually `typedef struct _win_st WINDOW;` and your definition here conflicts.